src/Stratis.Bitcoin/P2P/PeerDiscoveryLoop.cs

```diff
@@ -106,47 +114,57 @@ public void DiscoverPeers(IConnectionManager connectionManager)
         }
 
         /// <summary>
-        /// See <see cref="DiscoverPeers"/>.
+        /// See <see cref="DiscoverPeers"/>. This loop deals with discovery from DNS seeds and seed nodes as opposed to peers.
         /// </summary>
-        private async Task DiscoverPeersAsync()
+        private async Task DiscoverFromDnsSeedsAsync()
         {
             var peersToDiscover = new List<IPEndPoint>();
-            List<PeerAddress> foundPeers = this.peerAddressManager.PeerSelector.SelectPeersForDiscovery(1000).ToList();
-            peersToDiscover.AddRange(foundPeers.Select(p => p.Endpoint));
 
-            if (peersToDiscover.Count == 0)
+            if (this.peerAddressManager.Peers.Count == 0 || !this.peerAddressManager.Peers.Select(a => !a.Attempted).Any())
             {
-                // On normal circumstances the dns seeds are attempted only once per node lifetime.
-                if (this.isSeedAndDnsAttempted)
-                {
-                    this.logger.LogTrace("(-)[DNS_ATTEMPTED]");
-                    return;
-                }
-
                 this.AddDNSSeedNodes(peersToDiscover);
                 this.AddSeedNodes(peersToDiscover);
-                this.isSeedAndDnsAttempted = true;
-
-                if (peersToDiscover.Count == 0)
-                {
-                    this.logger.LogTrace("(-)[NO_ADDRESSES]");
-                    return;
-                }
-
-                peersToDiscover = peersToDiscover.OrderBy(a => RandomUtils.GetInt32()).ToList();
             }
             else
             {
-                // If all attempts have failed then attempt the dns seeds again.
-                if (!this.isSeedAndDnsAttempted && foundPeers.All(peer => peer.Attempted))
-                {
-                    peersToDiscover.Clear();
-                    this.AddDNSSeedNodes(peersToDiscover);
-                    this.AddSeedNodes(peersToDiscover);
-                    this.isSeedAndDnsAttempted = true;
-                }
+                this.logger.LogTrace("(-)[NOTHING_TO_DO]");
+                return;
+            }
+
+            if (peersToDiscover.Count == 0)
```

Comment:
This if will then only be applicable to checking if there are peers after we've added them from the dns and seed nodes.